src/app/pages/transaction-request/transaction-request.tsx

```diff
@@ -55,21 +59,25 @@ function TransactionRequestBase() {
   const { walletType } = useWalletType();
   const generateUnsignedTx = useGenerateUnsignedStacksTransaction();
   const ledgerNavigate = useLedgerNavigate();
+  const requestNetwork = useTransactionRequestNetworkState();
+  const networks = useNetworks();
+  const dispatch = useAppDispatch();
   const navigate = useNavigate();
+  // TODO: Remove - this is temporary for network refactor
+  useSetCurrentNetworkAtom();
 
   useRouteHeader(<PopupHeader />);
 
-  useEffect(() => void analytics.track('view_transaction_signing'), [analytics]);
-
-  const txRequest = useTransactionRequest();
-
-  // This exists to move away from the pattern where the search param is pull
-  // from an atom, rather than from the tooling provided by the app's router
-  useSetTransactionRequestAtom(txRequest);
+  useEffect(() => {
+    void analytics.track('view_transaction_signing'), [analytics];
+    if (!requestNetwork) return;
+    const reqNetworkId = findMatchingNetworkKey(requestNetwork, networks);
+    if (!reqNetworkId) return;
+    dispatch(requestActions.requestNetworkActivated(reqNetworkId));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
See https://github.com/hirosystems/stacks-wallet-web/pull/2660

This has been bothering me for a while, we shouldn't be adding all these eslint escapes, yet sometimes we need to.